compiler/rustc_hir_analysis/src/check/compare_method.rs

```diff
@@ -300,6 +300,17 @@ fn compare_predicate_entailment<'tcx>(
         return Err(emitted);
     }
 
+    // Finally, we need to check that the impl's args are well-formed given
+    // the hybrid param-env (impl + trait method where-clauses).
+    for arg in impl_sig.inputs_and_output {
+        ocx.register_obligation(traits::Obligation::new(
+            tcx,
+            cause.clone(),
+            param_env,
+            ty::Binder::dummy(ty::PredicateKind::WellFormed(arg.into())),
```

Comment:
:thinking: the only question I guess I have is whether or not we need to also check that the unnormalized impl arg is also WF...

Reply:
Done :)